nbdime/merging/chunks.py

```diff
@@ -93,47 +94,47 @@ def make_chunks(boundaries, diff0, diff1):
         k = boundaries[i+1] if i < nb-1 else j
         # Collect diff entries from each side
         # starting at beginning of this chunk
-        d0 = ()
-        while i0 < len(diff0) and diff0[i0].key == j:
-            d0 += (diff0[i0],)
-            i0 += 1
-        d1 = ()
-        while i1 < len(diff1) and diff1[i1].key == j:
-            d1 += (diff1[i1],)
-            i1 += 1
+        sub_diffs = []
+        for m, d in enumerate(diffs):
+            dis = ()
+            while i_diffs[m] < len(d) and d[i_diffs[m]].key == j:
+                dis += (d[i_diffs[m]],)
+                i_diffs[m] += 1
+            sub_diffs.append(dis)
         # Add non-empty chunks
-        if j < k or d0 or d1:
-            chunks.append((j, k, d0, d1))
+        if j < k or any(sub_diffs):
+            chunks.append((j, k, *sub_diffs))
```

Comment:
This use of * for tuple unpacking isn't available before python 3.5, I suggest doing: (j, k) + tuple(sub_diffs)